src/components/MultiGestureCanvas/usePanGesture.ts

```diff
@@ -0,0 +1,160 @@
+            clampedOffset,
+            isInHoriztontalBoundary,
+            isInVerticalBoundary,
+        };
+    }, [canvasSize.width, canvasSize.height]);
+
+    // We want to smoothly decay/end the gesture by phasing out the pan animation
+    // In case the content is outside of the boundaries of the canvas,
+    // we need to move the content back into the boundaries
+    const finishPanGesture = useWorkletCallback(() => {
+        // If the content is centered within the canvas, we don't need to run any animations
+        if (offsetX.value === 0 && offsetY.value === 0 && panTranslateX.value === 0 && panTranslateY.value === 0) {
+            return;
+        }
+
+        const {clampedOffset, isInHoriztontalBoundary, isInVerticalBoundary, horizontalBoundaries, verticalBoundaries} = getBounds();
+
+        // If the content is within the horizontal/vertical boundaries of the canvas, we can smoothly phase out the animation
+        // If not, we need to snap back to the boundaries
+        if (isInHoriztontalBoundary) {
+            // If the (absolute) velocity is 0, we don't need to run an animation
+            if (Math.abs(panVelocityX.value) !== 0) {
+                // Phase out the pan animation
+                offsetX.value = withDecay({
+                    velocity: panVelocityX.value,
+                    clamp: [horizontalBoundaries.min, horizontalBoundaries.max],
+                    deceleration: PAN_DECAY_DECELARATION,
+                    rubberBandEffect: false,
+                });
+            }
+        } else {
+            // Animated back to the boundary
+            offsetX.value = withSpring(clampedOffset.x, SPRING_CONFIG);
+        }
+
+        if (isInVerticalBoundary) {
+            // If the (absolute) velocity is 0, we don't need to run an animation
+            if (Math.abs(panVelocityY.value) !== 0) {
+                // Phase out the pan animation
+                offsetY.value = withDecay({
+                    velocity: panVelocityY.value,
+                    clamp: [verticalBoundaries.min, verticalBoundaries.max],
+                    deceleration: PAN_DECAY_DECELARATION,
+                });
+            }
+        } else {
+            // Animated back to the boundary
+            offsetY.value = withSpring(clampedOffset.y, SPRING_CONFIG);
+        }
+
+        // Reset velocity variables after we finished the pan gesture
+        panVelocityX.value = 0;
+        panVelocityY.value = 0;
+    });
+
+    const panGesture = Gesture.Pan()
+        .manualActivation(true)
+        .averageTouches(true)
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        .onTouchesMove((_evt, state) => {
+            // We only allow panning when the content is zoomed in
+            if (zoomScale.value <= 1 || isPagerSwiping.value) {
+                return;
+            }
+
+            state.activate();
+        })
+        .onStart(() => {
+            stopAnimation();
+        })
```

Comment:
This used to be `onBegin`. Is the change here intended?

Reply:
Yes. It makes no difference, just for consistency with the other gesture handlers